Bottom: f463d040ccc9a7cc12396fd91bb740b7bb9116cf
Top:    b5058ce8e86744331d78c831592e48543d77c44c
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-04-28 13:35:36 -0400

Refresh of mpc5200-dma-split

---

diff --git a/sound/soc/fsl/mpc5200_dma.c b/sound/soc/fsl/mpc5200_dma.c
index 068e04d..64255eb 100644
--- a/sound/soc/fsl/mpc5200_dma.c
+++ b/sound/soc/fsl/mpc5200_dma.c
@@ -148,7 +148,7 @@ int mpc5200_audio_dma_startup(struct psc_dma *psc_dma)
 }
 EXPORT_SYMBOL_GPL(mpc5200_audio_dma_startup);
 
-int mpc5200_audio_dma_hw_free(struct snd_pcm_substream *substream,
+static int pcm_dma_psc_free(struct snd_pcm_substream *substream,
 			   struct snd_soc_dai *dai)
 {
 	snd_pcm_set_runtime_buffer(substream, NULL);
@@ -163,7 +163,7 @@ EXPORT_SYMBOL_GPL(mpc5200_audio_dma_hw_free);
  * This function is called by ALSA to start, stop, pause, and resume the DMA
  * transfer of data.
  */
-int mpc5200_audio_dma_trigger(struct snd_pcm_substream *substream, int cmd,
+static int pcm_dma_psc_trigger(struct snd_pcm_substream *substream, int cmd,
 			   struct snd_soc_dai *dai)
 {
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
@@ -407,6 +407,7 @@ static struct snd_pcm_ops psc_dma_pcm_ops = {
 	.close		= psc_dma_pcm_close,
 	.ioctl		= snd_pcm_lib_ioctl,
 	.pointer	= psc_dma_pcm_pointer,
+	.trigger	= psc_dma_pcm_trigger
 };
 
 static u64 psc_dma_pcm_dmamask = 0xffffffff;
diff --git a/sound/soc/fsl/mpc5200_dma.h b/sound/soc/fsl/mpc5200_dma.h
index cf08a6c..16b4df2 100644
--- a/sound/soc/fsl/mpc5200_dma.h
+++ b/sound/soc/fsl/mpc5200_dma.h
@@ -73,14 +73,8 @@ struct psc_dma {
 
 int mpc5200_audio_dma_startup(struct psc_dma *psc_dma);
 
-int mpc5200_audio_dma_hw_free(struct snd_pcm_substream *substream,
-			   struct snd_soc_dai *dai);
-
 void mpc5200_audio_dma_shutdown(struct psc_dma *psc_dma);
 
-int mpc5200_audio_dma_trigger(struct snd_pcm_substream *substream, int cmd,
-			   struct snd_soc_dai *dai);
-
 extern struct snd_soc_platform mpc5200_audio_dma_platform;
 
 #endif /* __SOUND_SOC_FSL_MPC5200_DMA_H__ */
diff --git a/sound/soc/fsl/mpc5200_psc_i2s.c b/sound/soc/fsl/mpc5200_psc_i2s.c
index d03bf46..f45d269 100644
--- a/sound/soc/fsl/mpc5200_psc_i2s.c
+++ b/sound/soc/fsl/mpc5200_psc_i2s.c
@@ -141,8 +141,6 @@ static int psc_i2s_set_fmt(struct snd_soc_dai *cpu_dai, unsigned int format)
  */
 static struct snd_soc_dai_ops psc_i2s_dai_ops = {
 	.hw_params	= psc_i2s_hw_params,
-	.hw_free	= mpc5200_audio_dma_hw_free,
-	.trigger	= mpc5200_audio_dma_trigger,
 	.set_sysclk	= psc_i2s_set_sysclk,
 	.set_fmt	= psc_i2s_set_fmt,
 };
